################################################################################
SHORT DESCRIPTION: 
################################################################################
Increase command line in Linux real-mode boot header to 1024 bytes

################################################################################
LONG DESCRIPTION: 
################################################################################
Source: https://sourceforge.net/p/tboot/mailman/message/35574458/

256 bytes is awfully short, and 1024 matches the size of the temporary
buffer tboot uses to manipulate the command line.

Signed-off-by: Ed Swierk <eswierk@...>
---
 tboot/common/linux.c        | 4 ++--
 tboot/include/linux_defns.h | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

################################################################################
PATCHES 
################################################################################
Index: tboot-1.9.5/tboot/common/linux.c
===================================================================
--- tboot-1.9.5.orig/tboot/common/linux.c
+++ tboot-1.9.5/tboot/common/linux.c
@@ -120,8 +120,8 @@ bool expand_linux_image(const void *linu
 
     /* recommended layout
         0x0000 - 0x7FFF     Real mode kernel
-        0x8000 - 0x8FFF     Stack and heap
-        0x9000 - 0x90FF     Kernel command line
+        0x8000 - 0x8CFF     Stack and heap
+        0x8D00 - 0x90FF     Kernel command line
         for details, see linux_defns.h
     */
 
Index: tboot-1.9.5/tboot/include/linux_defns.h
===================================================================
--- tboot-1.9.5.orig/tboot/include/linux_defns.h
+++ tboot-1.9.5/tboot/include/linux_defns.h
@@ -207,7 +207,7 @@ typedef struct __attribute__ ((packed))
 
 #define REAL_KERNEL_OFFSET      0x0000
 #define BOOT_SECTOR_OFFSET      0x0200
-#define KERNEL_CMDLINE_OFFSET   0x9000
+#define KERNEL_CMDLINE_OFFSET   0x8D00
 #define REAL_END_OFFSET         0x9100
 
 #define REAL_MODE_SIZE          REAL_END_OFFSET - REAL_KERNEL_OFFSET
